### DISCUSSION TITLE
Release 0.5.11 (1.21.0.20)

### DISCUSSION THREAD

**Discussion by m-brisch** [Mojang/minecraft-editor#169] [Announcements]
_[Read the previous release notes here](https://github.com/Mojang/minecraft-editor/discussions/168)_  

---  
  

## Summary  

Hello! Super excited to share with you several big features building up toward our v0.6 release! We are now on 0.5.11 (1.21.0.20), as of Wednesday, April 3, 2024. 

As a reminder of Editor API support:   

- Editor Extensions [overview article](https://learn.microsoft.com/en-us/minecraft/creator/documents/editorextensionsintroduction) and [API reference documentation](https://aka.ms/EditorAPI) on learn.microsoft.com  
- [Starter Kit](https://github.com/Mojang/minecraft-editor-extension-starter-kit): Contains the compiler tools and various libraries required to get started making Editor Extensions  
- [Samples](https://github.com/Mojang/minecraft-editor-extension-samples) created by our team to jump-start your own Extension creations  

Additionally, if you're new to the Editor GitHub this week, we encourage you to join the Bedrock Add-Ons Discord [#Editor channel](https://discord.gg/TZnQKFwGpt)!  

 

---   

## New Feature: Block Hotbar + Updated Picker:
- At the top of the Editor there is now a Block Hotbar to easily swap between the blocks your tools use.
  - Click once or hit the corresponding key shortcut to select an element from the hotbar
  - Click again to open the block picker to customize
- Editor extensions can leverage the hotbar using `uiSession.extensionContext.blockPalette.getSelectedBlockType()`
  - NOTE: The current API is subject to change over the next few weeks to add some additional functionality

![image](https://github.com/Mojang/minecraft-editor/assets/29804144/859a9106-1d26-4f77-b1c1-ad3ea5185cf6)

![image](https://github.com/Mojang/minecraft-editor/assets/29804144/c039ff96-2a19-4c88-85bc-e6fcfd51b246)

## Updated Feature: Line Tool with widget support
We overhauled the line tool to take make it easier to visualize and edit before making changes to your world.

- Added CTRL+A to select all widgets in a line
- Added CTRL+D to deselect all widgets in a line
- Added continuous CTRL+D to continually remove the last segment from the line list until it's empty
- Added SHIFT+CTRL+D to completely clear all selected widget AND lines
- Added CTRL+F to fill
- Added CTRL+SHIFT+F to fill and leave the line intact
- Added DEL to fill the line with AIR
- Added SHIFT+DEL to fill and leave the line intact
- Added Keyboard movement (U/D/L/R/PG_UP/PG_DN) to manually nudge selected end points

![image](https://github.com/Mojang/minecraft-editor/assets/29804144/bb78c956-c5c8-4442-80fd-409d473875be)

## New Feature: Time-of-Day Settings Panel
We created a new settings panel to control the time-of-day within your world.

- Accessible from `World Options => Time of Day Settings`
- Use `Daylight Cycle` to toggle between time being active/inactive
- Use the `Presets` dropdown to snap to a time, or use `None` and the slider to pick a moment-in-time

![image](https://github.com/Mojang/minecraft-editor/assets/29804144/d2975927-93f9-4340-8225-73ea93104f88)

## Change Log   

- Updated 'Pause Actors' to 'Pause Entities'
- Added ‘Time of Day’ settings panel
- Updated block atlas file using different rendering technique
- Updated Line Tool with new functionality and widget support
- Added global block hotbar with updated block picker

 
---   

  

Thank you for being here! You're the best! ❤️

**Comment by MetalManeMc** [comment]
Love the detailed changelogs! ❤️
